components/Modals/BatchTransfer/index.tsx

```diff
@@ -0,0 +1,343 @@
+        setSelectedCoin(e)
+    }
+
+    const saveCurrency = () => {
+        if (selectedCoin.length > 5) {
+            toast.error('Maximum 5 currency', {
+                position: 'top-center',
+                autoClose: 3000,
+                hideProgressBar: false,
+                closeOnClick: true,
+                pauseOnHover: true,
+                draggable: true,
+                progress: undefined,
+                theme: toastTheme(isLight),
+                toastId: 2,
+            })
+        } else {
+            try {
+                for (let i = 0; i < selectedCoin.length; i += 1) {
+                    const currency = selectedCoin[i].value
+                    let isExist = inputCoin.some(
+                        (val) =>
+                            val?.split('@')[0] === currency &&
+                            val?.split('@')[1] !== ''
+                    )
+                    if (!isExist) {
+                        toast.error('Please fill all value', {
+                            position: 'top-center',
+                            autoClose: 3000,
+                            hideProgressBar: false,
+                            closeOnClick: true,
+                            pauseOnHover: true,
+                            draggable: true,
+                            progress: undefined,
+                            theme: toastTheme(isLight),
+                            toastId: 2,
+                        })
+                        throw Error
+                    }
+                }
+                for (let i = 0; i < inputCoin.length; i += 1) {
+                    const amount = inputCoin[i]?.split('@')[1]
+                    const input_ = Number(amount)
+                    if (isNaN(input_)) {
+                        toast.error(t('The input is not a number.'), {
+                            position: 'bottom-right',
+                            autoClose: 2000,
+                            hideProgressBar: false,
+                            closeOnClick: true,
+                            pauseOnHover: true,
+                            draggable: true,
+                            progress: undefined,
+                            theme: toastTheme(isLight),
+                            toastId: 3,
+                        })
+                        throw new Error()
+                    }
+                }
+                setSavedCurrency(true)
+            } catch {
+                setSavedCurrency(false)
+            }
+            console.log(inputCoin)
+        }
+    }
+
+    const resetState = () => {
+        setInputCoin([])
+        setSelectedCoin([])
+        setSavedCurrency(false)
+    }
+
+    const handleSubmit = async () => {
+        setIsLoading(true)
+        const zilpay = new ZilPayBase()
+        let params: any = []
+        const addr: TransitionParams = {
+            vname: 'addr',
+            type: 'ByStr20',
+            value: resolvedInfo?.addr,
```

Comment:
this is valid only for /funds

Reply:
Hi @tralkan, then what we should send for /balances?

Reply:
the recipient of the batch transfer

Reply:
I see, so we need recipient input for /balances. Let me add this on notion

Reply:
Yes, since it is a withdrawal. It can be a recipient address or NFT domain name @devwistkey .